lib/facter/resolvers/solaris/processors.rb

```diff
@@ -28,6 +28,13 @@ def parse_output(output)
             @fact_list[:physical_count] = output.scan(/chip_id .*/).uniq.size
             @fact_list[:speed] = output.scan(/current_clock_Hz .*/).first.gsub(/[a-zA-z\s]+/, '').to_i
             @fact_list[:models] = output.scan(/brand .*/).map { |elem| elem.gsub(/brand(\s+)/, '') }
+            calculate_threads_cores(output)
+          end
+
+          def calculate_threads_cores(output)
+            @fact_list[:threads_per_core] = @fact_list[:logical_count] / (output.scan(/core_id .*/).uniq.size -
```

Comment:
@BobosilaVictor I think it would be better to be more specific with the regex expression, such as `/\Wcore_id .*/` or `/\score_id .*/` which tries to match either a non-alphanumeric character or a whitespace before `core_id`.

Also, if we need to scan for `core_id` twice, my preference would be to add it to a `@fact_list` variable since regex calls are expensive.